src/hooks/useTxBatchTrades.tsx

```diff
@@ -39,51 +39,48 @@ export function useTxBatchTrades(
-  const ordersFoundInTx = orders?.length
 
-  const _fetchTxTrades = useCallback(
-    async (network: Network, _txHash: string): Promise<void> => {
-      setIsLoading(true)
-      setError('')
+  const _fetchTxTrades = useCallback(async (network: Network, _txHash: string, orders: Order[]): Promise<void> => {
+    setIsLoading(true)
+    setError('')
 
-      try {
-        const { transfers, trades } = await getTradesAndTransfers(network, _txHash)
-        const _accounts = Object.fromEntries(await getTradesAccount(network, _txHash, trades, transfers))
-        const orderIds = orders?.map((order) => order.owner) || []
-        const transfersWithKind: Transfer[] = transfers.reduce(
-          (acc, transfer) =>
-            !orderIds.includes(transfer.from) && !orderIds.includes(transfer.to) ? [...acc, transfer] : acc,
-          [],
-        )
+    try {
+      const { transfers, trades } = await getTradesAndTransfers(network, _txHash)
+      const _accounts = Object.fromEntries(await getTradesAccount(network, _txHash, trades, transfers))
+      const orderIds = orders.map((order) => order.owner) || []
+      const transfersWithKind: Transfer[] = transfers.reduce(
+        (acc, transfer) =>
+          !orderIds.includes(transfer.from) && !orderIds.includes(transfer.to) ? [...acc, transfer] : acc,
+        [],
+      )
 
-        orders?.forEach((order) => {
-          const { owner, kind } = order
-          transfersWithKind.push(
-            ...transfers.filter((t) => [t.from, t.to].includes(owner)).map((transfer) => ({ ...transfer, kind })),
-          )
-        })
+      orders.forEach((order) => {
+        const { owner, kind } = order
+        transfersWithKind.push(
+          ...transfers.filter((t) => [t.from, t.to].includes(owner)).map((transfer) => ({ ...transfer, kind })),
+        )
+      })
 
-        setErc20Addresses(transfers.map((transfer: Transfer): string => transfer.token))
-        setTxBatchTrades({ trades, transfers: transfersWithKind })
-        setAccounts(_accounts)
-      } catch (e) {
-        const msg = `Failed to fetch tx batch trades`
-        console.error(msg, e)
-        setError(msg)
-      } finally {
-        setIsLoading(false)
-      }
-    },
-    [orders],
-  )
+      setErc20Addresses(transfers.map((transfer: Transfer): string => transfer.token))
+      setTxBatchTrades({ trades, transfers: transfersWithKind })
+      setAccounts(_accounts)
+    } catch (e) {
+      const msg = `Failed to fetch tx batch trades`
+      console.error(msg, e)
+      setError(msg)
+    } finally {
+      setIsLoading(false)
+    }
+  }, [])
 
   useEffect(() => {
-    if (!networkId || !ordersFoundInTx) {
+    if (!networkId || !orders?.length) {
       return
     }
 
-    _fetchTxTrades(networkId, txHash)
-  }, [_fetchTxTrades, networkId, ordersFoundInTx, txHash])
+    _fetchTxTrades(networkId, txHash, orders)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why is this necessary?

Reply:
Good point, maybe it is not necessary, 
to avoid the rule breaking I can think of 2 things.
1. Continue to use `orders` and not execute `_fetchTrades` if it is **noLength/undfined**.
2. Or, pass it to `_fetchTrades` and stop it if there is **noLength/undfined** of `orders`.

Reply:
@ramirotw @henrypalacios The thing is not related to  it is undefined or not. The problem is the hook is executed twice once orders is populated (Since it is an array of objects, we would need to check for deep comparison in order not to execute it unnecessarily). So I'm using the orders length to prevent this.

Reply:
I meant the eslint rule

Reply:
I understand @matextrem ! As you use `orders` as a dependency but it needs the **length** reference. 

To avoid disabling **eslint** we could pull the hook [`usePreviousLastValueData()`](https://github.com/cowprotocol/explorer/blob/develop/src/apps/explorer/components/SummaryCardsWidget/VolumeChart/VolumeChart.tsx#L213), to reuse it in cases like this.

Reply:
I finally found the problem! The problem was orders array was really changing. It seems like the first time we get the orders and the second time it is populated with `buyToken` and `sellToken` data so it caused the hook to re-render. Since we are not using those properties, I discarded them and use `previousValue` hook to look for any changes in order to improve the performance along with avoid unnecessary requests to tenderly API